src/python/kdtree.cpp

```diff
@@ -41,5 +68,86 @@ void define_kdtree(py::module& m) {
       },
       py::arg("pt"),
       py::arg("k"),
-      "Search the k-nearest neighbors. Returns a pair of indices and squared distances.");
-}
\ No newline at end of file
+      R"""(
+       Find the k nearest neighbors to a given point.
+
+       Parameters
+       ----------
+       pt : NDArray, shape (3,)
+           The input point.
+       k : int
+           The number of nearest neighbors to search for.
+
+       Returns
+       -------
+       k_indices : NDArray, shape (k,)
+           The indices of the k nearest neighbors in the point cloud.
+       k_sq_dists : NDArray, shape (k,)
+           The squared distances to the k nearest neighbors.
+     )""")
+    .def(
+      "batch_nns_search",
```

Comment:
I would like to rename this method to `batch_nearest_neighbor_search` for consistency.